akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/scaladsl/R2dbcSession.scala

```diff
@@ -7,12 +7,53 @@ package akka.projection.r2dbc.scaladsl
 import scala.collection.immutable
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
+import scala.concurrent.duration._
 
 import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.persistence.r2dbc.ConnectionFactoryProvider
+import akka.persistence.r2dbc.ConnectionPoolSettings
 import akka.persistence.r2dbc.internal.R2dbcExecutor
+import akka.projection.r2dbc.R2dbcProjectionSettings
 import io.r2dbc.spi.Connection
 import io.r2dbc.spi.Row
 import io.r2dbc.spi.Statement
+import org.slf4j.LoggerFactory
+
+object R2dbcSession {
+  private val log = LoggerFactory.getLogger(classOf[R2dbcSession])
+
+  private val logDbCallsDisabled = -1.millis
+
+  /**
+   * Runs the passed function in using a R2dbcSession with a new transaction. The connection is closed and the
+   * transaction is committed at the end or rolled back in case of failures.
+   */
+  def withSession[A](system: ActorSystem[_])(fun: R2dbcSession => Future[A]): Future[A] = {
+    withSession(system, connectionFactoryConfigPath(system))(fun)
+  }
+
+  def withSession[A](system: ActorSystem[_], connectionFactoryConfigPath: String)(
+      fun: R2dbcSession => Future[A]): Future[A] = {
+    val connectionFactoryProvider = ConnectionFactoryProvider(system)
+    val connectionFactory = connectionFactoryProvider.connectionFactoryFor(connectionFactoryConfigPath)
+    val poolSettings = new ConnectionPoolSettings(system.settings.config.getConfig(connectionFactoryConfigPath))
```

Comment:
We need the `closeCallsExceeding` setting. This can be compared with https://github.com/akka/akka-persistence-r2dbc/blob/43fc9bece6f3f0dc91d30d0041e9b430e9d4ea61/core/src/main/scala/akka/persistence/r2dbc/session/scaladsl/R2dbcSession.scala#L40-L44
which is using internal `r2dbc`. Too much internals to expose `ConnectionFactorySettings`.